composables/drop/massmint/useDropMassMintListing.ts

```diff
@@ -0,0 +1,41 @@
+import isEqual from 'lodash/isEqual'
+import nftEntitiesByIDs from '@/queries/subsquid/general/nftEntitiesByIDs.graphql'
+
+export default () => {
+  const { client } = usePrefix()
+  const { listNftByNftWithMetadata } = useListingCartModal({
+    clearItemsOnModalClose: true,
+  })
+
+  const { mintedNFTs } = storeToRefs(useDropStore())
+
+  const subscribeForNftsWithMetadata = (nftIds: string[]) => {
+    subscribeToNfts(nftIds, async (data) => {
+      const ids = data.map((item) => item.id)
+      const readyToFetch = isEqual(ids.sort(), nftIds.sort())
+
+      if (readyToFetch) {
+        const { data } = await useAsyncQuery<{
+          nftEntities: NFTWithMetadata[]
+        }>({
+          query: nftEntitiesByIDs,
+          variables: {
+            ids: nftIds,
+          },
+          clientId: client.value,
+        })
+
+        mintedNFTs.value = data.value.nftEntities
+      }
+    })
+  }
+
+  const listMintedNFTs = () => {
+    mintedNFTs.value.forEach(listNftByNftWithMetadata)
```

Comment:
forEach is not sutiable here since `listNftByNftWithMetadata` returns a Promise

you should use `Promise.all` or `Promise.allSettled()` probably


https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise#static_methods

Reply:
no need, listNftByNftWithMetadata does not return a promise